CSVSource/src/main/java/io/vantiq/extsrc/CSVSource/CSVCore.java

```diff
@@ -55,24 +55,27 @@ public void handleMessage(ExtensionServiceMessage message) {
             }
             
             oConfigHandler.configComplete = false;
-                        
-            CompletableFuture<Boolean> success = client.connectToSource();
-            
-            try {
-                if ( !success.get(10, TimeUnit.SECONDS) ) {
-                    if (!client.isOpen()) {
-                        log.error("Failed to connect to server url '{}'.", targetVantiqServer);
-                    } else if (!client.isAuthed()) {
-                        log.error("Failed to authenticate within 10 seconds using the given authentication data.");
-                    } else {
-                        log.error("Failed to connect within 10 seconds");
+
+            // Spin off a thread to handle doing the reconnect and error/log handling, this way the handler doesn't
+            // block the onMessage() handler in the ExtensionWebSocketListener from processing future messages
+            new Thread( () -> {
+                CompletableFuture<Boolean> success = client.connectToSource();
```

Comment:
Are there any issues with degenerate conditions where we could run out of threads here?  Do we need to have only one of these open at a time? Or is that already handled?  Some `configInProgress` or something?

It's rare, so probably not necessary...